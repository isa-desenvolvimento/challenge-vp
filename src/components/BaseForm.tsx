import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
