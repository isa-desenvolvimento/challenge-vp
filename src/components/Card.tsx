import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'

export const Card = ({
  header,
  content,
  footer,
  primary,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer role="card">
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
