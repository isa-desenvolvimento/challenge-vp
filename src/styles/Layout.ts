import styled from 'styled-components'

export const StyleContainer = styled.main`
  max-height: 100vh;

  display: grid;
  grid-template-rows: 4rem auto;
  grid-template-columns: 1rem auto;

  grid-template-areas:
    'nav nav'
    'sidebar contentBox';
  text-align: center;

  font-family: ${({ theme }) => theme.fontFamily};

  overflow: hidden;

  @media screen and (max-width: 600px) {
    overflow-y: scroll;
  }
`
