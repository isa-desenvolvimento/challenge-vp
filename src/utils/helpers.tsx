import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
