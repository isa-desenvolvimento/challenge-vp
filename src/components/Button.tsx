import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
