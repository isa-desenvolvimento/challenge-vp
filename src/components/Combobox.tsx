import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
