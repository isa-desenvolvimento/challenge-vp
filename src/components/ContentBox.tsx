import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
