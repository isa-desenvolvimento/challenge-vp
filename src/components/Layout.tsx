import { ContentBox } from '@/components/ContentBox'
import { NavBar } from '@/components/NavBar'
import { Sidebar } from '@/components/SideBar'
import { StyleContainer } from '@/styles/Layout'
import { ChildrenProps } from '@/types/global'

export const Layout = ({ children }: ChildrenProps) => {
  return (
    <StyleContainer role="layout">
      <NavBar />
      <Sidebar />
      <ContentBox> {children} </ContentBox>
    </StyleContainer>
  )
}
