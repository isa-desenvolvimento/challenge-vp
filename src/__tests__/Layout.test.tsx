import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { Layout } from '@/components/Layout'

describe('<Layout />', () => {
  beforeEach(() =>
    renderWithProvider(
      <Layout>
        <h1>Test Layout</h1>
      </Layout>
    )
  )

  it('should render text Layout', () => {
    expect(screen.getByRole(/layout/i).textContent).toContain('Simulação')
  })

  it('should render the styled-components Layout', () => {
    expect(screen.getByRole(/layout/i)).toHaveStyle({
      'grid-area': 'contentBox',
      overflow: 'hidden',
      width: '100%',
      height: '100%',
      background: '#eff0f6',
    })
  })

  it('should render snapshot', () => {
    expect(screen.getByRole(/layout/i)).toMatchSnapshot()
  })
})
