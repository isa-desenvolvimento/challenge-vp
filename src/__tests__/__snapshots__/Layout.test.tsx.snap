// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render snapshot 1`] = `
.c9 {
  grid-area: contentBox;
  overflow: scroll;
  width: 100%;
  height: 100%;
  background: #eff0f6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  padding: 1rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #36a4a8;
  font-size: 120;
}

.c2:hover {
  background-color: #eceff5;
  color: #36a4a8;
}

.c2:active {
  background-color: #eceff5;
  color: #36a4a8;
}

.c1 {
  grid-area: nav;
  overflow: hidden;
  background-color: #eceff5;
  box-shadow: 0px 1px 5px #cec8c8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
  z-index: 1;
}

.c5 {
  text-align: justify;
  margin: 1rem;
}

.c6 {
  font-weight: bold;
  color: #60647C;
}

.c7 {
  color: #ddd;
}

.c3 {
  background-color: transparent;
  box-sizing: border-box;
  border-left: 1px solid #ddd;
  margin-left: auto;
  padding: 1rem;
  height: 75%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: block;
  border-radius: 50%;
}

.c8 {
  grid-area: sidebar;
  width: 20px;
  height: 95vh;
  background: #446996;
}

.c0 {
  max-height: 100vh;
  display: grid;
  grid-template-rows: 4rem auto;
  grid-template-columns: 1rem auto;
  grid-template-areas: 'nav nav' 'sidebar contentBox';
  text-align: center;
  font-family: 'Nunito';
  overflow: hidden;
}

@media screen and (max-width:600px) {

}

@media screen and (max-width:600px) {
  .c8 {
    height: auto;
  }
}

@media screen and (max-width:600px) {
  .c0 {
    overflow-y: scroll;
  }
}

<main
  class="c0"
  role="layout"
>
  <div
    class="c1"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4IiBoZWlnaHQ9IjY0IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <noscript />
      <img
        alt="Logo Vamos Parcelar"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <a
      class="c2"
      role="link-item"
    >
      
      Simulação
    </a>
    <div
      class="c3"
      role="link-avatar"
      width="50"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="imagem do perfil"
          class="c4"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
      <div
        class="c5"
      >
        <h5
          class="c6"
        >
          Julia Arruda Fonseca
        </h5>
        <h5
          class="c7"
        >
          Gerente
        </h5>
        <h5
          class="c7"
        />
      </div>
    </div>
  </div>
  <div
    class="c8"
    role="sidebar"
  />
  <main
    class="c9"
    role="contentBox"
  >
     
     
    <h1>
      Test Layout
    </h1>
     
  </main>
</main>
`;
